src/Behat/Behat/Context/Snippet/ContextSnippet.php

```diff
@@ -20,32 +20,23 @@
  */
 final class ContextSnippet implements Snippet
 {
-    /**
-     * @var StepNode
-     */
-    private $step;
-    /**
-     * @var string
-     */
-    private $template;
-    /**
-     * @var string
-     */
-    private $contextClass;
+    private StepNode $step;
+
+    private string $template;
+
+    private string $contextClass;
+
     /**
      * @var string[]
      */
-    private $usedClasses;
+    private array $usedClasses;
 
     /**
      * Initializes definition snippet.
      *
-     * @param StepNode $step
-     * @param string   $template
-     * @param string   $contextClass
      * @param string[] $usedClasses
      */
-    public function __construct(StepNode $step, $template, $contextClass, array $usedClasses = array())
+    public function __construct(StepNode $step, string $template, string $contextClass, array $usedClasses = [])
```

Comment:
Since we are modernizing this code, maybe use constructor property promotion?

Reply:
I agree :)

I was thinking of proposing this kind of modification in another PR, across the entire codebase.